### PR TITLE
#Fix 1007 Drag and Drop functionality for pdf and policy files has be…

### PR DESCRIPTION
**Pull request by ArtemKiriutin**
Added Drag and Drop functionality for pdf and policy files.
Fixed failing policy result message color and text.
closed https://github.com/veraPDF/veraPDF-library/issues/1007
closed https://github.com/veraPDF/veraPDF-library/issues/1027

### CODE REVIEW

**Comment by carlwilson** [comment]
Hi @ArtemKiriutin thanks very much for this, it's been requested before and a PR is useful. We're a bit busy on a couple of fronts right now but we'll test, review and merge / get back to you before the year is out.

File: gui/src/main/java/org/verapdf/gui/ValidateWorker.java [hunk 1]
```diff
@@ -178,4 +185,28 @@ private void writeHtmlReport() {
 			this.htmlReport = null;
 		}
 	}
+
+	private int countFailedPolicyJobs(File xmlReport) throws ParserConfigurationException, SAXException, IOException {
+		SAXParserFactory factory = SAXParserFactory.newInstance();
```
Comment (BezrukovM) [inline gui/src/main/java/org/verapdf/gui/ValidateWorker.java]: use XPath here

File: gui/src/main/java/org/verapdf/gui/utils/ValidateWorkerSummary.java [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+import org.verapdf.processor.reports.BatchSummary;
+import org.verapdf.processor.reports.FeaturesBatchSummary;
+import org.verapdf.processor.reports.MetadataRepairBatchSummary;
+import org.verapdf.processor.reports.ValidationBatchSummary;
+
+
+/**
+ * @author Artem Kiriutin
+ */
+public class ValidateWorkerSummary implements BatchSummary {
```
Comment (BezrukovM) [inline gui/src/main/java/org/verapdf/gui/utils/ValidateWorkerSummary.java]: make this summary an inner class of Validate worker and do not implement Batch summary

File: gui/src/main/java/org/verapdf/gui/CheckerPanel.java [hunk 3]
```diff
@@ -844,9 +846,80 @@ public ProcessingTypeRenderer() {
...
+		//Panel in which dropTargetListener is registered
+		private CheckerPanel panel;
+
+		private int acceptableFilesCount;
+
+		private String[] acceptableExtensions;
+
+
+		public PanelDropTargetListener(CheckerPanel panel, int acceptableFilesCount, String...acceptableExtensions) {
+			this.panel = panel;
```
Comment (BezrukovM) [inline gui/src/main/java/org/verapdf/gui/CheckerPanel.java]: as this class is not static - we do not need to add panel as an argument

File: gui/src/main/java/org/verapdf/gui/CheckerPanel.java [hunk 4]
```diff
@@ -844,9 +845,74 @@ public ProcessingTypeRenderer() {
...
 	}
+
+	public class PanelDropTargetListener extends DropTargetAdapter {
+		private final Logger LOGGER = Logger.getLogger(PanelDropTargetListener.class.getCanonicalName());
+
+		private static final String UNSUPPORTED_FLAVOUR = "Unsupported flavour error.";
+		private static final String CASTING_ERROR = "Casting transfer to files type error.";
+
+		//Panel in which dropTargetListener is registered
+		private CheckerPanel panel;
```
Comment (BezrukovM) [inline gui/src/main/java/org/verapdf/gui/CheckerPanel.java]: is this field in use?

File: gui/src/main/java/org/verapdf/gui/ValidateWorker.java [hunk 5]
```diff
@@ -44,14 +53,15 @@
...
-class ValidateWorker extends SwingWorker<BatchSummary, Integer> {
-
+class ValidateWorker extends SwingWorker<ValidateWorker.ValidateWorkerSummary, Integer> {
 	private static final Logger logger = Logger.getLogger(ValidateWorker.class.getCanonicalName());
 
 	private static final String ERROR_IN_OPEN_STREAMS = "Can't open stream from PDF file or can't open stream to temporary XML report file"; //$NON-NLS-1$
 	private static final String ERROR_IN_PROCESSING = "Error during the processing"; //$NON-NLS-1$
 	private static final String ERROR_IN_CREATING_TEMP_FILE = "Can't create temporary file for XML report"; //$NON-NLS-1$
 	private static final String ERROR_IN_OBTAINING_POLICY_FEATURES = "Can't obtain enabled features from policy files"; //$NON-NLS-1$
+	private static final String POLICY_REPORT = "policyReport";
```
Comment (BezrukovM) [inline gui/src/main/java/org/verapdf/gui/ValidateWorker.java]: remove

**Pull request merged**